Scripts/Modules/Highlight/HighlightModule.cs

```diff
@@ -7,35 +7,86 @@ namespace UnityEditor.Experimental.EditorVR.Modules
 {
 	sealed class HighlightModule : MonoBehaviour
 	{
-		static readonly Vector3 k_HighlightScaleIncrease = Vector3.one * 0.0125f;
+		[SerializeField]
+		Material m_DefaultHighlightMaterial;
+
+		[SerializeField]
+		Material m_LeftHighlightMaterial;
 
 		[SerializeField]
-		private Material m_HighlightMaterial;
+		Material m_RightHighlightMaterial;
+
+		readonly Dictionary<Transform, HashSet<GameObject>> m_Highlights = new Dictionary<Transform, HashSet<GameObject>>();
+		readonly HashSet<GameObject> m_DefaultHighlights = new HashSet<GameObject>();
+		readonly Dictionary<Node, HashSet<Transform>> m_NodeMap = new Dictionary<Node, HashSet<Transform>>();
+
+		public Color leftColor
+		{
+			get { return m_LeftHighlightMaterial.color; }
+		}
 
-		private readonly Dictionary<GameObject, int> m_HighlightCounts = new Dictionary<GameObject, int>();
+		public Color rightColor
+		{
+			get { return m_RightHighlightMaterial.color; }
+		}
 
 		void LateUpdate()
 		{
-			foreach (var go in m_HighlightCounts.Keys)
+			foreach (var highlight in m_Highlights)
 			{
-				if (go == null)
-					continue;
-
-				foreach (var m in go.GetComponentsInChildren<MeshFilter>())
+				var highlights = highlight.Value;
+				foreach (var go in highlights)
 				{
-					var highlightTransform = m.transform;
-					Matrix4x4 highlightScaleIncreaseMatrix = Matrix4x4.TRS(highlightTransform.position, highlightTransform.rotation, highlightTransform.lossyScale + k_HighlightScaleIncrease);
-
-					if (m.sharedMesh == null)
+					if (go == null)
 						continue;
 
-					for (var i = 0; i < m.sharedMesh.subMeshCount; i++)
-						Graphics.DrawMesh(m.sharedMesh, highlightScaleIncreaseMatrix, m_HighlightMaterial, m.gameObject.layer, null, i);
+					var node = Node.LeftHand;
+					foreach (var kvp in m_NodeMap)
+					{
+						if (kvp.Value.Contains(highlight.Key))
+						{
+							node = kvp.Key;
+							break;
+						}
+					}
+
+					var material = node == Node.LeftHand ? m_LeftHighlightMaterial : m_RightHighlightMaterial;
+
+					HighlightObject(go, material);
 				}
 			}
+
+			foreach (var go in m_DefaultHighlights)
+			{
+				HighlightObject(go, m_DefaultHighlightMaterial);
+			}
+		}
+
+		static void HighlightObject(GameObject go, Material material)
+		{
+			foreach (var m in go.GetComponentsInChildren<MeshFilter>())
+			{
+				if (m.sharedMesh == null)
+					continue;
+
+				for (var i = 0; i < m.sharedMesh.subMeshCount; i++)
+					Graphics.DrawMesh(m.sharedMesh, m.transform.localToWorldMatrix, material, m.gameObject.layer, null, i);
+			}
+		}
+
+		public void AddRayOriginForNode(Node node, Transform rayOrigin)
+		{
+			HashSet<Transform> set;
+			if (!m_NodeMap.TryGetValue(node, out set))
+			{
+				set = new HashSet<Transform>();
+				m_NodeMap[node] = set;
+			}
+
+			set.Add(rayOrigin);
 		}
 
-		public void SetHighlight(GameObject go, bool active)
+		public void SetHighlight(GameObject go, Transform rayOrigin, bool active)
```

Comment:
Can we address #36 while we're it as well and provide a method to disable selection highlight for a single object? It could be a part of this method or a separate method. Essentially we would want it to be added to an ignored list or "highlight hidden" list.